packages/editor-ui/src/components/TimeAgo.vue

```diff
@@ -1,76 +1,67 @@
-<template>
-	<span :title="convertDate">
-		{{ format }}
-	</span>
-</template>
-
-<script lang="ts">
-import type { LocaleFunc } from 'timeago.js';
+<script lang="ts" setup>
 import { format, register } from 'timeago.js';
 import { convertToHumanReadableDate } from '@/utils/typesUtils';
-import { defineComponent } from 'vue';
-import { mapStores } from 'pinia';
+import { computed, onBeforeMount } from 'vue';
 import { useRootStore } from '@/stores/n8nRoot.store';
+import { useI18n } from '@/composables/useI18n';
 
-export default defineComponent({
-	name: 'TimeAgo',
-	props: {
-		date: {
-			type: String,
-			required: true,
-		},
-		capitalize: {
-			type: Boolean,
-			default: false,
-		},
-	},
-	computed: {
-		...mapStores(useRootStore),
-		defaultLocale(): string {
-			return this.rootStore.defaultLocale;
-		},
-		format(): string {
-			const text = format(this.date, this.defaultLocale);
+type Props = {
+	date: string;
+	capitalize?: boolean;
+};
 
-			if (!this.capitalize) {
-				return text.toLowerCase();
-			}
+const props = withDefaults(defineProps<Props>(), {
+	capitalize: false,
+});
```

Comment:
Do we prefer this style over:

```ts
defineProps({
  capitalize: {
    type: Boolean,
    default: false
  }
})
```
?

Reply:
This style has better typing support out of the box. I would say so.